### PR TITLE
Create a filter for actions without olives

### PR DESCRIPTION
**Pull request by apmasell**
Adds a new filter type to find actions that have no currently know olive.

- [X] Updates Changelog
- [X] Updates developer documentation

### CODE REVIEW

**Comment by avarsava** [comment]
Thank you for the updates, however this still does not compile:

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile (default-testCompile) on project shesmu-pluginapi: Compilation failure: Compilation failure: 
[ERROR] /home/avarsava/Workspace/shesmu/shesmu-pluginapi/src/test/java/ca/on/oicr/gsi/shesmu/plugin/TextFilterParseTest.java:[132,49] <anonymous ca.on.oicr.gsi.shesmu.plugin.TextFilterParseTest$1> is not abstract and does not override abstract method orphaned() in ca.on.oicr.gsi.shesmu.plugin.filter.ActionFilterBuilder
[ERROR] /home/avarsava/Workspace/shesmu/shesmu-pluginapi/src/test/java/ca/on/oicr/gsi/shesmu/plugin/TextFilterParseTest.java:[161,49] <anonymous ca.on.oicr.gsi.shesmu.plugin.TextFilterParseTest$2> is not abstract and does not override abstract method orphaned() in ca.on.oicr.gsi.shesmu.plugin.filter.ActionFilterBuilder
[ERROR] /home/avarsava/Workspace/shesmu/shesmu-pluginapi/src/test/java/ca/on/oicr/gsi/shesmu/plugin/TextFilterParseTest.java:[191,49] <anonymous ca.on.oicr.gsi.shesmu.plugin.TextFilterParseTest$3> is not abstract and does not override abstract method orphaned() in ca.on.oicr.gsi.shesmu.plugin.filter.ActionFilterBuilder
`

**Comment by apmasell** [comment]
Corrected. I've also merged with this with #1227 so there can be a policy based on liveness.

**Review by callunity — APPROVED** [review]

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Server.java [hunk 1]
```diff
@@ -262,17 +262,17 @@ public Server(int port, FileWatcher fileWatcher) throws IOException, ParseExcept
...
 
     /**
-     * When fetch()ing a format, try the name of the format on:
-     * - AnnotatedInputFormatDefinition's Stream of known formats, which includes -input and -remote files
-     * - PluginManager, which will stream its known formats (installed plugins) and try fetching from each of those
-     * - ActionProcessor, which offers the 'shesmu' input format
-     * We try every format and flatMap the results together in order to offer the ability to specify the same input
-     * format multiple ways at the same time. E.g., a cerberus plugin may be configured AND a cerberus_fp-remote file
-     * (a RemoteJsonSource in AnnotatedInputFormatDefinition) may be present at the same time. We
+     * When fetch()ing a format, try the name of the format on: - AnnotatedInputFormatDefinition's
```
Comment (avarsava) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Server.java:265]: oh no the formatting

**Comment by avarsava** [comment]
Deployed this feature and tried it out on my local shesmu that has no actions. 
- The 'Orphaned' filter causes the action view to display the sad olive and "JSON.parse: unexpected end of data at line 1 column 1 of the JSON data"
- the 'OliveOrphaned' tile cannot be dismissed by clicking on it, and there's no 'x' in the corner either
- Add Filter -> Tags says 'Failed to load: 400 Bad Request' only when the OliveOrphaned tile is present in the filter bar

**Comment by apmasell** [comment]
Big yikes. Not sure what happened, but I've rebase and fixed it.

**Pull request merged**